crates/nu-command/src/platform/kill.rs

```diff
@@ -0,0 +1,149 @@
+use nu_engine::CallExt;
+use nu_protocol::engine::{Command, EngineState, Stack};
+use nu_protocol::{ast::Call, span};
+use nu_protocol::{
+    Category, Example, IntoPipelineData, PipelineData, ShellError, Signature, Spanned, SyntaxShape,
+    Value,
+};
+use std::process::{Command as CommandSys, Stdio};
+
+#[derive(Clone)]
+pub struct Kill;
+
+impl Command for Kill {
+    fn name(&self) -> &str {
+        "kill"
+    }
+
+    fn usage(&self) -> &str {
+        "Kill a process using the process id."
+    }
+
+    fn signature(&self) -> Signature {
+        let signature = Signature::build("kill")
+            .required(
+                "pid",
+                SyntaxShape::Int,
+                "process id of process that is to be killed",
+            )
+            .rest("rest", SyntaxShape::Int, "rest of processes to kill")
+            .switch("force", "forcefully kill the process", Some('f'))
+            .switch("quiet", "won't print anything to the console", Some('q'))
+            .category(Category::Platform);
+
+        if cfg!(windows) {
+            return signature;
+        }
+
+        signature.named(
+            "signal",
+            SyntaxShape::Int,
+            "signal decimal number to be sent instead of the default 15 (unsupported on Windows)",
+            Some('s'),
+        )
+    }
+
+    fn run(
+        &self,
+        engine_state: &EngineState,
+        stack: &mut Stack,
+        call: &Call,
+        _input: PipelineData,
+    ) -> Result<PipelineData, ShellError> {
+        let pid: i64 = call.req(engine_state, stack, 0)?;
+        let rest: Vec<i64> = call.rest(engine_state, stack, 1)?;
+        let force: bool = call.has_flag("force");
+        let signal: Option<Spanned<i64>> = call.get_flag(engine_state, stack, "signal")?;
+        let quiet: bool = call.has_flag("quiet");
+
+        let mut cmd = if cfg!(windows) {
+            let mut cmd = CommandSys::new("taskkill");
+
+            if force {
+                cmd.arg("/F");
+            }
+
+            cmd.arg("/PID");
+            cmd.arg(pid.to_string());
+
+            // each pid must written as `/PID 0` otherwise
+            // taskkill will act as `killall` unix command
+            for id in &rest {
+                cmd.arg("/PID");
+                cmd.arg(id.to_string());
+            }
+
+            cmd
+        } else {
+            let mut cmd = CommandSys::new("kill");
+            if force {
+                if let Some(Spanned {
+                    item: _,
+                    span: signal_span,
+                }) = signal
+                {
+                    return Err(ShellError::IncompatibleParameters {
+                        left_message: "force".to_string(),
+                        left_span: call.get_named_arg("force").unwrap().span,
```

Comment:
Ahh, we don't have `unwrap()` prevention yet. I'll try to add that to the CI.

You might want to refactor to remove those. I was also wondering if there was a way to use the `.get_flag(..)` of the style of line 56 above

Reply:
Actually, don't worry about the `unwrap()`. It looks like a bunch crept in since we didn't have the clippy flag to check for them.

Reply:
> I was also wondering if there was a way to use the .get_flag(..) of the style of line 56 above

`get_flag` only works for arguments with values like `--signal 123`, and returns spanned value (e.g. 123). It does not work for switches (will always return None), since there is no value and no corresponding span.

The only alternative I can think of without `get_named_arg` would be just returning `ShellError::IncompatibleParametersSingle` with `call.head` span, but that's less precise.

Reply:
Sure, using a helper is fine then.